src/main/java/com/lambdaschool/oktafoundation/config/OktaAuthSecurityConfig.java

```diff
@@ -35,25 +41,26 @@ protected void configure(HttpSecurity http)
 				.permitAll()
 
 				// *** NOTE AUTHENTICATED CAN READ USERS!!! PATCHES are handled in UserService
-				.antMatchers("/users/**")
-				.authenticated()
-				// *** Handled at UseremailService Level
+				.antMatchers("/users/**", "/okta/**")
+				.permitAll()
 				.antMatchers("/useremails/**")
-				.authenticated()
+//				.authenticated()
+				// *** Handled at UseremailService Level
+				.permitAll()
 				.antMatchers("/modules/**", "/programs/**", "/courses/**", "/students/**")
-				.authenticated()
+				.permitAll()
 				.antMatchers("/teachers/**")
-				.authenticated()
+				.permitAll()
 				.antMatchers(HttpMethod.GET, "/courses/**", "/modules/**", "/students/**")
 				.permitAll()
 				.antMatchers(HttpMethod.POST, "/courses/**", "modules/**", "/teachers/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 				.antMatchers(HttpMethod.PATCH, "/courses/**", "/modules/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 				.antMatchers(HttpMethod.PUT, "/courses/**", "/modules/**", "/students/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 				.antMatchers(HttpMethod.DELETE, "/courses/**", "/modules/**", "/students/**", "/teachers/**")
-				.hasAnyRole("ADMIN", "TEACHER")
+				.permitAll()
 
```

Comment:
I don't think you meant to upload this version of this file. If so, just let me know why and I'll approve.